### PR TITLE
Use `Vector*` component-wise `min/max/clamp` functions where applicable

### PR DESCRIPTION
**Pull request by AThousandShips**
Improves readability and safety, separated off the `clamp` changes as they might be less clear

### CODE REVIEW

File: platform/windows/display_server_windows.cpp [hunk 1]
```diff
@@ -1632,8 +1631,7 @@ void DisplayServerWindows::window_set_current_screen(int p_screen, WindowID p_wi
 		Size2i wsize = window_get_size(p_window);
 		wpos += srect.position;
 
-		wpos.x = CLAMP(wpos.x, srect.position.x, srect.position.x + srect.size.width - wsize.width / 3);
-		wpos.y = CLAMP(wpos.y, srect.position.y, srect.position.y + srect.size.height - wsize.height / 3);
+		wpos = wpos.clamp(srect.position, srect.position + srect.size - wsize / 3);
```
Comment (bruvzg) [inline platform/windows/display_server_windows.cpp:1634]: `platform/macos/display_server_macos.mm` have the same `wpos` code as well (and similar part in `update_mouse_wrap`, but it's using NSPoint, so probably should not be changed).

Comment (AThousandShips) [reply platform/windows/display_server_windows.cpp:1634]: Ah, good point, realized I didn't include `.mm` files, will add those in a bit

**Review by Geometror — APPROVED** [review]
Went through each replacement and didn't notice any obvious mistake.
Nice work!

**Comment by AThousandShips** [comment]
Thank you! Will do a new search for the current state otherwise this is good and ready

**Comment by akien-mga** [comment]
Thanks!

**Comment by AThousandShips** [comment]
Thank *you*!

**Pull request merged**